test/run

```diff
@@ -0,0 +1,268 @@
+    echo "ERROR[exec /bin/bash -c "${run_cmd}"] Expected '${expected}', got '${out}'"
+    return 1
+  fi
+  out=$(docker exec $(cat ${cid_file}) /bin/sh -ic "${run_cmd}" 2>&1)
+  if ! echo "${out}" | grep -q "${expected}"; then
+    echo "ERROR[exec /bin/sh -ic "${run_cmd}"] Expected '${expected}', got '${out}'"
+    return 1
+  fi
+  return 0
+}
+
+function test_scl_usage() {
+  scl_usage "ruby --version" "ruby ${RUBY_VERSION}."
+  check_result "$?"
+}
+
+function test_npm_functionality() {
+  info "Testing npm availibility"
+  ct_npm_works
+  check_result "$?"
+}
+
+function handle_test_case_result() {
+    local output_format
+    output_format="%s %s $1\n"
+    shift
+    local test_msg
+    if [ $TESTCASE_RESULT -eq 0 ]; then
+      test_msg="[PASSED]"
+    else
+      test_msg="[FAILED]"
+      TESTSUITE_RESULT=1
+    fi
+    printf -v test_short_summary "${output_format}" "${test_short_summary}" "${test_msg}" $@
+}
+
+function run_all_tests() {
+  local APP_NAME="$1"
+  for test_case in $TEST_SET; do
+    info "Running test $test_case ... "
+    $test_case
+    check_result $?
+    local test_msg
+    handle_test_case_result "for '%s' %s" "${APP_NAME}" "$test_case"
+    [ -n "${FAIL_QUICKLY:-}" ] && {
+      cleanup "${APP_NAME}"
+      return 1
+    }
+  done;
+
+  cleanup "${APP_NAME}"
+}
+
+function test_application() {
+  # Verify that the HTTP connection can be established to test application container
+  run_test_application &
+
+  # Wait for the container to write it's CID file
+  wait_for_cid
+  check_result $?
+}
+
+function test_scl_variables_in_dockerfile() {
+  if [ "$OS" == "rhel7" ] || [ "$OS" == "centos7" ]; then
+    # autocleanup only enabled here as only the following tests so far use it
+    CID_FILE_DIR=$(mktemp -d)
+    ct_enable_cleanup
+
+    info "Testing variable presence during \`docker exec\`"
+    ct_check_exec_env_vars
+    check_result $?
+
+    info "Checking if all scl variables are defined in Dockerfile"
+    ct_check_scl_enable_vars
+    check_result $?
+ fi
+}
+
+function test_from_dockerfile() {
+  rootdir=$(readlink -zf ${image_dir}/..)
+  dockerfile="Dockerfile${1:-}"
+  info "Check building using a $dockerfile"
+  ct_test_app_dockerfile $rootdir/examples/from-dockerfile/$dockerfile 'https://github.com/sclorg/rails-ex.git' 'Welcome to your Rails application on OpenShift' app-src
+  check_result $?
+  handle_test_case_result "test_from_dockerfile for %s" "${dockerfil}"
+}
+
+pushd ${test_dir}
+if [ -d db-test-app ]; then
+  rm -rf db-test-app
+fi
+git clone git://github.com/openshift/ruby-hello-world db-test-app
+popd
+
+test_from_dockerfile
+test_from_dockerfile ".s2i"
+
+for server in ${WEB_SERVERS[@]}; do
+  cid_file=$(mktemp -u --suffix=.cid)
+
+  # Since we built the candidate image locally, we don't want S2I attempt to pull
+  # it from Docker hub
+  s2i_args="--pull-policy=never"
+
+  run_s2i_build "${server}"
+  check_result $?
+  handle_test_case_result "build of '%s' app" "${server}"
```

Comment:
This is not needed. The whole print is used in row 260.

Reply:
I'm aware that this adds one more line into the total tests results, but without this, a potential error during image build would not be visible that well, it would be visible in the first test case for the build -- that could make the error investigation harder.